### ISSUE TITLE
bugs/redundant-existence-check does not consider "with"

### ISSUE THREAD

**Issue by asleire** [StyraInc/regal#733]
I constructed this [play](https://play.openpolicyagent.org/p/OV98jjTI7M) based on an unexpected lint error in a test in a real life project.

```
rule := input

rule_test if {
	not rule.foo with input as {}

	x := rule.foo with input as {"foo": "bar"}
	x == "bar"
}
```

Regal considers the `not rule.foo` line redundant because we use `rule.foo` in the later line, but these two lines have different outputs of `rule.foo` because of the `with` statement

It's worth noting that in this example I am performing two separate tests in a single rule and that I could split them into two rules instead which would remove this warning. I'm not sure there is any valid use case outside of tests where one would get this false positive

**Comment by anderseknert** [comment]
Ah, yes, that's a good catch! I'll try to have it fixed before the v0.22.0 release going out today 👍

**Comment by anderseknert** [comment]
PR up! Now just waiting for @charlieegan3 to wake up and review :)
